core/math/color.h

```diff
@@ -176,16 +176,16 @@ struct _NO_DISCARD_ Color {
 
 	_FORCE_INLINE_ Color srgb_to_linear() const {
 		return Color(
-				r < 0.04045f ? r * (1.0 / 12.92) : Math::pow((r + 0.055f) * (float)(1.0 / (1 + 0.055)), 2.4f),
-				g < 0.04045f ? g * (1.0 / 12.92) : Math::pow((g + 0.055f) * (float)(1.0 / (1 + 0.055)), 2.4f),
-				b < 0.04045f ? b * (1.0 / 12.92) : Math::pow((b + 0.055f) * (float)(1.0 / (1 + 0.055)), 2.4f),
+				r < 0.04045f ? r * (float)(1.0 / 12.92) : Math::pow((r + 0.055f) * (float)(1.0 / 1.055), 2.4f),
+				g < 0.04045f ? g * (float)(1.0 / 12.92) : Math::pow((g + 0.055f) * (float)(1.0 / 1.055), 2.4f),
+				b < 0.04045f ? b * (float)(1.0 / 12.92) : Math::pow((b + 0.055f) * (float)(1.0 / 1.055), 2.4f),
```

Comment:
Does it really make a significant difference to do these operations as double before casting, instead of float from the start?

Reply:
The linear to srgb just uses float btw.

Reply:
I would also keep 1.0 + 0.055 separate, like 1.0 / 2.4. This matches the terms (A, Gamma) of the equation: https://en.m.wikipedia.org/wiki/SRGB
Compilers will optimize this anyway, better keep the code close to how we represent it in math (see how A is also used in other places).

Reply:
> Does it really make a significant difference to do these operations as double before casting, instead of float from the start?

If this is between `(float)(1.0 / 12.92)` and `(1.0f / 12.92f)`, probably not a great deal (it depends if you do the reverse elsewhere, and want to get best accuracy, in case these two give different results), but the old version should be replaced to prevent promotion of the calculation to 64 bit.

Reply:
Yeah I would suggest doing `1.0f / 12.92f` directly.

Reply:
I just checked, `1.0f / 12.92f` is equal to `1.0 / 12.92`, but doing `(1.0 / (1.0 + 0.055))` with float literals would be one bit off. It doesn't matter much, but yes let's only use the double literals where it makes a difference.